storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -4121,7 +4236,43 @@ public boolean isTopologyNameAllowed(String name) throws AuthorizationException,
             throw new RuntimeException(e);
         }
     }
-    
+
+    @Override
+    public SupervisorAssignments getSupervisorAssignments(String node) throws AuthorizationException, TException {
+        try {
+            if(isLeader()) {
+                SupervisorAssignments supervisorAssignments = new SupervisorAssignments();
+                supervisorAssignments.set_storm_assignment(assignmentsForNode(stormClusterState.assignmentsInfo(), node));
+                return supervisorAssignments;
+            }
+        } catch (Exception e) {
+            //when not leader just return nil which will cause client to get an unknown error
```

Comment:
Why not leaving log message even if we could swallow the exception? That may help investigating the issue when Nimbus is malfunctioning.

Reply:
Friendly reminder: this comment is not addressed.

Reply:
I have added log.debug message.